investment-app/TRINV.Api/TRINV.Application/Queries/Notification/GetAllNotificationsQuery.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace TRINV.Application.Queries.Notification;
+
+using Domain.Entities;
+using Interfaces;
+using MediatR;
+using Shared.Business.Exceptions;
+using Shared.Business.Extension;
+using Shared.Business.Utilities;
+
+public record GetAllNotificationsQuery : IRequest<OperationResult<List<Notification>>>;
+
+internal class GetAllNotificationsQueryHandler : IRequestHandler<GetAllNotificationsQuery, OperationResult<List<Notification>>>
+{
+    readonly IRepository<Notification> _repository;
+
+    public GetAllNotificationsQueryHandler(IRepository<Notification> repository)
+    {
+        _repository = repository;
+    }
+
+    public async Task<OperationResult<List<Notification>>> Handle(GetAllNotificationsQuery request, CancellationToken cancellationToken)
+    {
+        var operationResult = new OperationResult<List<Notification>>();
+
+        var notifications = await _repository.GetAllAsync(cancellationToken);
+
+        if (!notifications.Any())
```

Comment:
Why we need to throw exception when there is no notifications ?

Reply:
Done.